### PR TITLE
Phase2-hgx129 Extend the topology/geometry codes for HFNose

### PR DESCRIPTION
**Pull request by bsunanda**
Some changes are needed in some of the xml files to make these work. This will be done once the previous PR's are merged

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/ForwardGeometry/test/HFNoseGeometryTester.cc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  ~HFNoseGeometryTester() override;
+
+  void beginJob() override {}
+  void analyze(edm::Event const& iEvent, edm::EventSetup const&) override;
+  void endJob() override {}
+  
+private:
+  void doTestWafer(const HGCalGeometry* geom);
+  
+  std::string    name;
```
Comment (ianna) [inline Geometry/ForwardGeometry/test/HFNoseGeometryTester.cc]: name_ ?

**Review by ianna — CHANGES_REQUESTED** [review]

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ianna** [comment]
+1

File: Geometry/CaloTopology/src/HGCalTopology.cc [hunk 2]
```diff
@@ -25,6 +26,11 @@ HGCalTopology::HGCalTopology(const HGCalDDDConstants& hdcons,
     subdet_     = (ForwardSubdetector)(det);
     kHGeomHalf_ = sectors_*layers_;
     types_      = 2;
+  } else if (det < (int)(DetId::HGCalEE)) {
```
Comment (kpedro88) [inline Geometry/CaloTopology/src/HGCalTopology.cc]: Why not just check `det==DetId::Forward` here?

Comment (bsunanda) [reply Geometry/CaloTopology/src/HGCalTopology.cc]: The value of det is the subdetector type which happens to be DetId::Forward. That is just a coincidence. I am not sure if that could be used.

Comment (kpedro88) [reply Geometry/CaloTopology/src/HGCalTopology.cc]: So `det` is not actually a parameter containing the `DetId::det()` field, but only matches up for HGCal?

**Comment by bsunanda** [comment]
Hi Kevin

May be I shall use the set as detector type and not sub detector type and change code accordingly. That will make things similar to the case of HGCAL. Cheers            Sunanda
________________________________

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ianna** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Pull request merged**